.gitignore

```diff
@@ -103,4 +103,10 @@ ENV/
 .mypy_cache/
 
 # Project specific
-tests/plugins/test.db
\ No newline at end of file
+tests/plugins/test.db
+
+# pycharm
+.idea/
+
+# pre-commit automatic linter
+.pre-commit-config.yaml
```

Comment:
why not include the file in git?

Reply:
I'm not sure how popular `pre-commit` really is and didn't want to pollute the top level directory with things that few people use. So, I'm a little hesitant. What do you say?

Reply:
I like `pre-commit` and don't regard the as pollution but as advertisement for a great tool.

Since we are now strict regarding formatting it might save new contributors some time if we offer a great development enviromnent